File: 198. House Robber/House Robber.java
class Solution {
    public int rob(int[] nums) {
        int rob1 = 0;
        int rob2 = 0;

        for (int index = 0; index < nums.length; ++index) {
            int temp = Math.max(rob1 + nums[index], rob2);
            rob1 = rob2;
            rob2 = temp;
        }

        return rob2;
    }
}

File: 198. House Robber/House Robber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function (nums) {
  if (nums.length === 1) {
    return nums[0];
  }

  let rob1 = 0;
  let rob2 = 0;
  for (let value of nums) {
    const temp = Math.max(rob1 + value, rob2);
    rob1 = rob2;
    rob2 = temp;
  }

  return rob2;
};
